lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_extract_keys_from_strings_files.rb

```diff
@@ -0,0 +1,119 @@
+                                         UI.user_error!("`source_parent_dir` should be a path to an existing directory, but found `#{value}`.") unless File.directory?(value)
+                                         UI.user_error!("`source_parent_dir` should contain at least one `.lproj` subdirectory, but `#{value}` does not contain any.") if Dir.glob('*.lproj', base: value).empty?
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :source_tablename,
+                                       env_name: 'FL_IOS_EXTRACT_KEYS_FROM_STRINGS_FILES_SOURCE_TABLENAME',
+                                       description: 'The basename of the `.strings` file (without the extension) to extract the keys and translations from for each locale',
+                                       type: String,
+                                       default_value: 'Localizable'),
+          FastlaneCore::ConfigItem.new(key: :target_original_files,
+                                       env_name: 'FL_IOS_EXTRACT_KEYS_FROM_STRINGS_FILES_TARGET_ORIGINAL_FILES',
+                                       description: 'The path to the `<base-locale>.lproj/<target-tablename>.strings` file(s) for which we want to extract the keys to. ' \
+                                        + 'Those files, containing the originals (typical `en` or `Base` locale), will be used to determine which keys to extract from the `source_tablename`, ' \
+                                        + 'and the target file path(s) in which to extract the translations will be determined from those path, writing them in the file with the same basename ' \
+                                        +' in each of the other `*.lproj` sibling folders of those target paths',
+                                       type: Array,
+                                       verify_block: proc do |values|
+                                         UI.user_error!('`target_original_files` must contain at least one path to an original `.strings` file.') if values.empty?
+                                         values.each do |v|
+                                           UI.user_error!("Path `#{v}` (found in `target_original_files`) does not exist.") unless File.exist?(v)
+                                           UI.user_error! "Expected `#{v}` (found in `target_original_files`) to be a path ending in a `*.lproj/*.strings`." unless File.extname(v) == '.strings' && File.extname(File.dirname(v)) == '.lproj'
+                                         end
+                                       end),
+        ]
+      end
+
+      def self.return_type
+        # Describes what type of data is expected to be returned
+        # see RETURN_TYPES in https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/action.rb
+        nil
+      end
+
+      def self.return_value
+        # Freeform textual description of the return value
+      end
```

Comment:
Yeah I always wonder about that as well when I create a new action.

But given most of us do a copy/paste of an old action to get a starting point when creating a new one, and also because every time I add those if I copied this from an old action that didn't have them:
 - I often then forget to re-introduce them if they end up being useful
 - Or even forget to just _consider_ and pause and ponder if my new action should have a return value or not in the first place (while having them around from the copied action makes you wonder about it which is a good thing imho)
 - And when I decide they are missing but should be added to my new action, I have to search for both the name of those, but also their role (e.g. "wait, is `return_type` supposed to be a freeform text? a Ruby type like `Array`? A symbol like `:array`?" and "is `return_value` supposed to be some symbol, or the actual value of a variable? Or just freeform text describing it?"), and it's easier to have them around and empty but documented rather than having to dig back into fastlane's own source code to find the parent class and retro-engineer their role and values